### ISSUE TITLE
Add more options to branch.protect() function

### ISSUE THREAD

**Issue by anderslindho** [python-gitlab/python-gitlab#1355]
The `branch.protect()` function currently supports two arguments:
```
developers_can_push: bool
developers_can_merge: bool
```
per https://python-gitlab.readthedocs.io/en/stable/gl_objects/branches.html

Since gitlab allows for more granularity than this, I would propose adding the arguments:
```
maintainers_can_push: bool
maintainers_can_merge: bool
```

The documentation appears to support this by specifying a higher access level (60): https://docs.gitlab.com/ce/api/protected_branches.html#protected-branches-api

**Comment by JohnVillalovos** [comment]
Have you looked at:

https://python-gitlab.readthedocs.io/en/stable/gl_objects/protected_branches.html

**Comment by JohnVillalovos** [comment]
Interestingly it looks like our code is calling an undocumented API. Maybe it was previously documented.

https://docs.gitlab.com/ce/api/branches.html

https://github.com/python-gitlab/python-gitlab/blob/96d2805b5bf372cb79c2b7db5c1e499c41e477c1/gitlab/v4/objects/branches.py#L21-L43

**Comment by JohnVillalovos** [comment]
It appears that branch.protect() was likely a v3 API thing. I'm guessing v4 still supports it but it is undocumented. I have not personally tried it.

https://gitlab.com/gitlab-org/gitlab-foss/-/blob/8-16-stable/doc/api/branches.md

**Comment by anderslindho** [comment]
I had not seen the first link you provided - I will try out that approach.

I have tried the `branch.protect()` function and can verifiy that it at least works.

**Comment by anderslindho** [comment]
I've tried this alternative approach now (`project.protectedbranches.create()`), and it doesn't seem to work as intended - nothing updated in the UI and it appeared as if I could still push (albeit that part is a little bit trickier to confirm as I don't know the intention behind the `60` status level; me being owner of the repo might still qualify here).

**Comment by JohnVillalovos** [comment]
Might be something like this:

```
    print("Deleting any existing projected branch info for:", args.branch)
    try:
        project.protectedbranches.delete(args.branch)
    except gitlab.exceptions.GitlabDeleteError:
        pass
    print("Protecting the branch:", args.branch)
    p_branch = project.protectedbranches.create(
        {
            "name": args.branch,
            "allowed_to_push": [{"access_level": gitlab.NO_ACCESS}],
        }
    )
    print("Protected branch:", pprint.pformat(vars(p_branch)))
```

**Comment by anderslindho** [comment]
That worked brilliantly - thank you very much! 🙏 

So my mistake was either:
- that it expects a list as argument (seems unlikely due to the examples in the documentation)
- that I typed in some minor error somewhere; but what's odd then is that everything seemed to work on the CLI-side (`project.protectedbranches.list()` showed that I had created new entries).

Either way, there doesn't seem to be any reason to keep this issue open any longer. Thank you for the help!
